komga/src/main/kotlin/org/gotson/komga/interfaces/rest/SeriesController.kt

```diff
@@ -323,57 +324,77 @@ class SeriesController(
     } ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
 
   @ApiResponse(content = [Content(schema = Schema(type = "string", format = "binary"))])
-  @GetMapping(value = ["{seriesId}/thumbnail"], produces = [MediaType.IMAGE_JPEG_VALUE])
+  @GetMapping(value = ["{seriesId}/thumbnail", "{seriesId}/thumbnail/{thumbnailId}"], produces = [MediaType.IMAGE_JPEG_VALUE])
   fun getSeriesThumbnail(
     @AuthenticationPrincipal principal: KomgaPrincipal,
-    @PathVariable(name = "seriesId") seriesId: String
+    @PathVariable(name = "seriesId") seriesId: String,
+    @PathVariable(name = "thumbnailId") thumbnailId: String?
   ): ByteArray {
     val series = seriesRepository.getLibraryId(seriesId) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
     if (!principal.user.canAccessLibrary(series)) throw ResponseStatusException(HttpStatus.FORBIDDEN)
 
-    val thumbnail = seriesLifecycle.getThumbnailBytes(seriesId, principal.user.id)
+    val thumbnail = if (thumbnailId != null) {
+      seriesLifecycle.getThumbnailBytesById(seriesId, thumbnailId)
```

Comment:
i suppose this would be to get the thumbnail image by its ID, for example to display all the available thumbnails in a dialog?

Reply:
Yeah, it will get the image by id, and like you said to be shown in a dialog together with `getSeriesThumbnails` to get which thumbnails are available for a series